core/src/main/java/org/wso2/testgrid/infrastructure/InfrastructureCombinationsProvider.java

```diff
@@ -50,14 +54,15 @@
 
     private static final Logger logger = LoggerFactory.getLogger(InfrastructureCombinationsProvider.class);
 
-    public Set<InfrastructureCombination> getCombinations() throws TestGridDAOException {
+    public Set<InfrastructureCombination> getCombinations(TestgridYaml testgridYaml) throws TestGridDAOException {
         Set<InfrastructureValueSet> valueSets = new InfrastructureParameterUOW().getValueSet();
         if (logger.isDebugEnabled()) {
-            logger.debug("Retrieved value-set from database: " + valueSets);
+            logger.debug(String.format("Retrieved value-set from database: %s", valueSets));
         }
-        Set<InfrastructureCombination> infrastructureCombinations = getCombinations(valueSets);
+        Set<InfrastructureCombination> infrastructureCombinations = getCombinations(
+                filterInfrastructures(valueSets, testgridYaml));
         if (logger.isDebugEnabled()) {
-            logger.debug("Generated set of infrastructure combinations: " + infrastructureCombinations);
+            logger.debug(String.format("Generated set of infrastructure combinations: %s", infrastructureCombinations));
```

Comment:
Shall we have the string concatenation with `+` here? AFAIK, `String#format` is slower since it first parses the input with regular expressions and then fills in the parameters. Concatenation with `+` is optimized by javac and it uses StringBuilder#append

Reply:
If you are looking for performance only I believe that using string concatenation is the most efficient approach. But IMO if we are looking for readability the String.format thing is the much clearer.  This code is executed when debug is enabled, therefore shall we use String.format instead of string concatenation. WDYT?  :)